src/tests/ftest/util/general_utils.py

```diff
@@ -304,3 +310,796 @@ def verify_rebuild(pool, log, to_be_rebuilt, object_qty, record_qty, errors=0):
+    task = task_self()
+
+    log.info("Checking for pool data on %s", nodeset)
+    for file in file_list:
+        task.run(
+            "test -e /mnt/daos/{}; echo $?".format(file), nodes=nodeset)
+        for buffer, node_list in task.iter_buffers():
+            if buffer == "0":
+                actual += len(node_list)
+            else:
+                nodes = NodeSet.fromlist(node_list)
+                log.error("%s: /mnt/daos/%s not found", nodes, file)
+    return expect == actual
+
+
+class CallbackHandler(object):
+    """Defines a callback method to use with DaosApi class methods."""
+
+    def __init__(self, delay=1):
+        """Create a CallbackHandler object.
+
+        Args:
+            delay (int, optional): number of seconds to wait in between
+                checking if the callback() method has been called.
+                Defaults to 1.
+        """
+        self.delay = delay
+        self.rc = None
+        self.obj = None
+        self._called = False
+
+    def callback(self, event):
+        """Return an event from a DaosApi class method.
+
+        Args:
+            event (CallbackEvent): event returned by the DaosApi class method
+        """
+        # Get the return code and calling object from the event
+        self.rc = event.event.ev_error
+        self.obj = event.obj
+
+        # Indicate that this method has being called
+        self._called = True
+
+    def wait(self):
+        """Wait for this object's callback() method to be called."""
+        # Reset the event return code and calling object
+        self.rc = None
+        self.obj = None
+
+        # Wait for the callback() method to be called
+        while not self._called:
+            print(" Waiting ...")
+            sleep(self.delay)
+
+        # Reset the flag indicating that the callback() method was called
+        self._called = False
+
+
+class TestParameter(object):
+    """A class for test parameters whose values are read from a yaml file."""
+
+    def __init__(self, value, default=None):
+        """Create a TestParameter object.
+
+        Args:
+            value (object): intial value for the parameter
+            default (object, optional): default value. Defaults to None.
+        """
+        self.value = value if value is not None else default
+        self.default = default
+
+    def __str__(self):
+        """Conver this object into a string.
+
+        Returns:
+            str: the string version of the parameter's value
+
+        """
+        return str(self.value)
+
+    def set_value(self, name, test, path):
+        """Set the value for the paramter from the test case's yaml file.
+
+        Args:
+            name (str): name of the value in the yaml file
+            test (Test): avocado Test object to use to read the yaml file
+            path (str): yaml path where the name is to be found
+        """
+        self.value = test.params.get(name, path, self.default)
+
+
+class TestDaosApiBase(object):
```

Comment:
(pylint-too-few-public-methods) Too few public methods (1/2)

Reply:
Shall we consider disabling this pylon error?

Reply:
Yes, thanks